PurchasesCoreSwift/Public/PurchaserInfo.swift

```diff
@@ -0,0 +1,326 @@
+//
+//  PurchaserInfo.swift
+//  PurchasesCoreSwift
+//
+//  Created by Madeline Beyl on 7/9/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+
+@objc(RCPurchaserInfo) public class PurchaserInfo: NSObject {
+
+    /// Entitlements attached to this purchaser info
+    @objc public let entitlements: EntitlementInfos
+
+    /// All *subscription* product identifiers with expiration dates in the future.
+    @objc public var activeSubscriptions: Set<String> { activeKeys(dates: expirationDatesByProduct) }
+
+    /// All product identifiers purchases by the user regardless of expiration.
+    @objc public var allPurchasedProductIdentifiers: Set<String> {
+        return self.nonConsumablePurchases.union(expirationDatesByProduct.keys)
```

Comment:
`pod lib lint` complains about this one and line 121, because we're using `nonConsumablePurchases`, but it's deprecated in favor of `nonSubscriptionTransactions`. I think we should be able to get the same behavior using `nonSubscriptionTransactions`, right?

Reply:
given a `PurchaserInfo` object, would this value ever change? if not, then maybe we can make it a `let` and set the value in the initializer

Reply:
sure, seems safe enough...

on the second comment -- i guess computed properties are computed for every `get`, but thoughts on making it a lazy variable instead? could probably do that with `latestExpirationDate` and `activeSubscriptions` too...or is there some advantage to having it set in init?

Reply:
for `activeSubscriptions` and others that compare against the current date, we have to keep them as computed properties since there's a chance that you'd get a different result when calling it at different times

Reply:
🤦 yep good catch